lib/rex/proto/mssql/client.rb

```diff
@@ -83,6 +83,8 @@ def map_compile_os_to_platform(server_info)
             platform = Msf::Platform::Windows.realname
           elsif os_data.match?('win')
             platform = Msf::Platform::Windows.realname
+          elsif os_data.match?(/\bon\b\s+(\w+)/)
```

Comment:
I'm just wondering was there a reason I might be missing for the matchers not just happening in the `detect_platform_and_arch` method you added in you're previous PR; [here](https://github.com/rapid7/metasploit-framework/pull/19112/files#diff-35b776d59b859a11793a463c4885eb146743c460ea717a29b958d7ad9d4c996dR115).

I did this in my [PR](https://github.com/rapid7/metasploit-framework/pull/19080/files#diff-8e9d347dd0cc409fe7f98bdd38e2846c657839c5b161b6e358f7caf43c2ffac9R216), not sure if it's better or worse, but my thinking was we match the values there and pass those `platform` and `arch` values like the method suggests, then just have a default with the method here. If that makes sense 😄

Reply:
I think that makes sense, especially to be consistent, I'll swap that up!